dts/arm/nxp/nxp_lpc55S6x_common.dtsi

```diff
@@ -198,6 +198,22 @@
 		status = "disabled";
 		#address-cells = <1>;
 		#size-cells = <0>;
+		hs_lspi_slave0: spi-slave@0 {
+			reg = <0x0>;
+			status = "disabled";
+		};
+		hs_lspi_slave1: spi-slave@1 {
+			reg = <0x1>;
+			status = "disabled";
+		};
+		hs_lspi_slave2: spi-slave@2 {
+			reg = <0x2>;
+			status = "disabled";
+		};
+		hs_lspi_slave3: spi-slave@3 {
+			reg = <0x3>;
+			status = "disabled";
+		};
```

Comment:
Could you please try to explain in plain words, what you are trying to fix with these changes? It is still unclear to me.
The approach you have taken here is not how our shield support is designed to work (and not how any other board within the tree allows SPI slaves to be configured in an overlay).

Reply:
The driver that is used by this shield, enc28j60 uses device tree to determin it's slave id.
`.spi_slave = DT_INST_REG_ADDR(0),`

Having the old definition, it will always defines the slave id as 0, which is incorrect and should be defined at board level.
```
eth_click@0 {...
		reg = <0>;
	};
```

This is why I changed to:
```
// shield level
eth_click: &mikrobus_spi_slave { ... }

// board level
mikrobus_spi_slave: &hs_lspi_slave1 {
};
```

I found no other way or examples in Zephyr on how to get the SPI bus ID. The only other example is a legacy definition and spi_loopback test which uses Kcofig `CONFIG_SPI_LOOPBACK_SLAVE_NUMBER`.

Reply:
Well, I think there are actually a bunch of spi slaves already in shields, and so far this didn't raise any issue. So I think you should try to comply with these.

I agree that when using several slaves on the same spi you'll hit issue of having multiple slaves with same address. This is not really a bug but rather a limitation in the way we're using overlays. 
This could be by-passed by defining arbitrary addresses per category of slaves (assuming you seldomly use 2 slaves of the same category), so all eth would be at address 0, bt slaves at address 1, screens at address 2, ....), but so far the need didn't arise.

Reply:
FYI: I have added PR #26323 which deals with a similar support issue for connecting the Mikroe ADC click to the LPCXpresso55S16. I have simply added a board-specific overlay to the shield definition. A similar solution should be possible for this PR.

Reply:
@henrikbrixandersen @erwango please review again, used @henrikbrixandersen's solution since i got no feedback.